tests/core/providers/test_auto_provider.py

```diff
@@ -25,6 +25,13 @@
 TEMP_DIR = gettempdir()
 
 
+# Need to have env var set before import or an error is raised
+os.environ['WEB3_INFURA_API_KEY'] = 'test'
+from web3.auto import (  # noqa E402 isort:skip
+    infura,
+)
+
```

Comment:
This also feels gross. I think it's my favorite option though because it keeps the weirdness contained to this test file. My next favorite option is just to have a normal environment variable set in CI and then make anyone developing have an environment variable set locally. Totally open to other options ¯\_(ツ)_/¯